clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -55,7 +55,9 @@
     private final int partitionLeaderEpoch;
     private final int writeLimit;
 
-    private volatile float estimatedCompressionRatio;
+    // Use a conservative estimate of the compression ratio. The producer overrides this using statistics
+    // from previous batches before appending any records.
+    private volatile float estimatedCompressionRatio = 1.0F;
```

Comment:
Nit: it probably doesn't matter here, but do we want to avoid the unnecessary volatile write as a general pattern?